pkg/discovery/discovery.go

```diff
@@ -278,6 +279,10 @@ func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConf
 	var discoErrors []error
 	var totalCapacityBlockBytes, totalCapacityFSBytes int64
 	for _, file := range files {
+		if config.DeviceNameMatching != "" && !strings.Contains(file, config.DeviceNameMatching) {
+			klog.Infof("file(%s) under(%s) does not match %s", file, config.MountDir, config.DeviceNameMatching)
+			continue
+		}
```

Comment:
how about making it generic?

1) rename `DeviceNameMatching` to `NamePattern` (filesystem mount points are also matched here)
2) use [filepath.Match](https://golang.org/pkg/path/filepath/#Match) to match on file name

Reply:
in your case, the pattern will be `nvm*`, the same pattern used in the shell.

Reply:
sounds good, will change the PR soon

Reply:
thanks, it's done. To make sure it's backward compatible, when `namePattern` is empty string, it would be invoke `filepath.Match`, so empty == "*" here.

Reply:
LGTM, btw can you add a helm value for this configuration (see [example](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/tree/master/helm#configurations), e.g. `classes.[n].fsType`)? you can do it in a separate PR.

Reply:
sure, thanks for the review